sdk/identity/azure_identity/README.md

```diff
@@ -1,77 +1,425 @@
+</!-->
+
+### Authenticate via development tools
+
+|Credential|Usage|Reference
+|-|-|-
+|[`AzureCliCredential`][cli_cred_ref]| Authenticates in a development environment with the Azure CLI. | [Azure CLI authentication](https://learn.microsoft.com/cli/azure/authenticate-azure-cli)
+<!-- ?: I don't believe we have a SecretCredential type yet>
+|[`AzureDeveloperCliCredential`][azd_cli_cred_ref]| Authenticates in a development environment with the Azure Developer CLI. | [Azure Developer CLI Reference](https://learn.microsoft.com/azure/developer/azure-developer-cli/reference)
+</!-->
+<!-- ?: I don't believe we have a SecretCredential type yet>
+|[`AzurePowerShellCredential`][powershell_cred_ref]| Authenticates in a development environment with the Azure PowerShell. | [Azure PowerShell authentication](https://learn.microsoft.com/powershell/azure/authenticate-azureps)
+</!-->
+<!-- ?: I don't believe we have a SecretCredential type yet>
+|[`VisualStudioCodeCredential`][vscode_cred_ref]| Authenticates as the user signed in to the Visual Studio Code Azure Account extension. | [VS Code Azure Account extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.azure-account)
+</!-->
+
+<!-- ?: I don't believe anything in this section is currently supported in the Rust Identity library yet, as we don't have an EnvironmentCredential type yet, and it doesn't look like DefaultAzureCredential includes the ClientCertificateCredential in it's chain. And I believe we are purposefully not implementing the username and password option as a security stance. >
+## Environment variables
+
+[DefaultAzureCredential][default_cred_ref] and [EnvironmentCredential][environment_cred_ref] can be configured with environment variables. Each type of authentication requires values for specific
+variables:
+
+### Service principal with secret
+
+|Variable name|Value
+|-|-
+|`AZURE_CLIENT_ID`|ID of a Microsoft Entra application
+|`AZURE_TENANT_ID`|ID of the application's Microsoft Entra tenant
+|`AZURE_CLIENT_SECRET`|one of the application's client secrets
+
+### Service principal with certificate
+
+|Variable name|Value|Required
+|-|-|-
+|`AZURE_CLIENT_ID`|ID of a Microsoft Entra application|X
+|`AZURE_TENANT_ID`|ID of the application's Microsoft Entra tenant|X
+|`AZURE_CLIENT_CERTIFICATE_PATH`|path to a PEM or PKCS12 certificate file including private key|X
+|`AZURE_CLIENT_CERTIFICATE_PASSWORD`|password of the certificate file, if any|
+|`AZURE_CLIENT_SEND_CERTIFICATE_CHAIN`|If `True`, the credential sends the public certificate chain in the x5c header of each token request's JWT. This is required for Subject Name/Issuer (SNI) authentication. Defaults to False. There's a [known limitation](https://github.com/Azure/azure-sdk-for-python/issues/13349) that async SNI authentication isn't supported.|
```

Comment:
Probably want to remove the known limitation comment if it isn't only a python issue.